plugins/extractors/kafka/kafka.go

```diff
@@ -202,6 +215,44 @@ func (e *Extractor) buildAsset(topicName string, numOfPartitions int) (*v1beta2.
 	}, nil
 }
 
+func (e *Extractor) readPartitions(ctx context.Context) ([]kafka.Partition, error) {
+	start := time.Now()
+
+	partitions, err := e.conn.ReadPartitions()
+	if err != nil {
+		errorCode := "UNKNOWN"
+		var kErr kafka.Error
+		if errors.As(err, &kErr) {
+			errorCode = strings.ReplaceAll(
+				strings.ToUpper(kErr.Title()), " ", "_",
+			)
+		}
+
+		e.clientDurn.Record(
+			ctx,
+			time.Since(start).Milliseconds(),
+			metric.WithAttributes(
+				attribute.String("kafka.broker", e.config.Broker),
+				attribute.String("kafka.error_code", errorCode),
+				attribute.Bool("success", false),
+			),
+		)
+
+		return nil, err
+	}
+
+	e.clientDurn.Record(
+		ctx,
+		time.Since(start).Milliseconds(),
+		metric.WithAttributes(
+			attribute.String("kafka.broker", e.config.Broker),
+			attribute.Bool("success", true),
+		),
+	)
```

Comment:
looks redundant, let's refactor it a bit?

Reply:
have refactored with a deferred function, please check.